document compatibility mode for oidc logout endpoint suppression (#1652)
keycloak/keycloak-documentation: upgrading/topics/keycloak/changes-19_0_0.adoc

The new {project_operator} now uses `StatefulSet` instead of `Deployment` for Keycloak deployments. There's no automated
migration in place given the Operator is a tech preview in this release. If you are using the new Operator with 18.0.z,
please make sure to back up, delete and recreate your Keycloak CR after the upgrade to 19.0.0.

= OpenID Connect Logout Prompt

> With Keycloak 18.0.0 the logout has been made compatible to the new OIDC specification, which changed the handling for the url parameters. To stay compatible with earlier versions a compatibility flag had been introduced. See link:{upgradingguide_link}#openid-connect-logout[{upgradingguide_name}] for further information for the backwards compatibility option, which allows your application to still use the old format for the url parameters.

Reviewer: ```suggestion
At Keycloak 18.0.0, the logout is now compatible with the new OIDC specification, which changed the handling for the url parameters. However, to also remain compatible with earlier versions, a compatibility flag is introduced. See the link:{upgradingguide_link}#openid-connect-logout[{upgradingguide_name}] for further information for the backwards compatibility option, which allows your application to still use the old format for the url parameters.
```

> While the url parameters can now be configured to be compatible, there is still one incompatibility with keycloak 17 an earlier. If the user does not provide an valid `idTokenHint` he will get a logout prompt instead of a successful logout redirect. Therefore a new compatibility flag `suppress-logout-confirmation-screen` has been introduced to suppress the logout screen. If enabled the logout screen will be disabled and never shown to the user, instead the user gets redirected (if the redirect url is valid).

Reviewer: ```suggestion
While the url parameters can now be configured to be compatible, there is still one incompatibility with keycloak 17 and earlier releases. If the user does not provide an valid `idTokenHint`, a logout prompt appears instead of a successful logout redirect. Therefore, a new compatibility flag `suppress-logout-confirmation-screen` is introduced to suppress the logout screen. If this flag is enabled, the logout screen will be disabled and never shown to the user. Instead, the user gets redirected (if the redirect url is valid).
```
